### PR TITLE
Subtitles downloading and secondary DownloadAssets

### PR DESCRIPTION
**Pull request by bensoftde**
Adds the option for secondary DownloadAssets in the `DownloadAsset` model and applies the new option for `Subtitles`. 

Major changes:
- secondary DownloadAsset handling is implemented on the `DownloadManager` layer
- adjusted `DownloadViewHelper` and `DownloadManager` to support it
- removed `getDownload(DownloadAsset)` method in `DownloadManager` which was accessed by `DownloadViewHelper` to provide updates on download status and replaced it with `getDownloadWrittenBytes(DownloadAsset)` and `getDownloadTotalBytes(DownloadAsset)`
- added `ARG_SHOW_NOTIFICATION` bundle field for `DownloadService` option to decide whether to show a download notification for a specific `DownloadAsset` (can be set in `DownloadAsset.showNotification`)
- `VideoHelper` uses downloaded subtitles if available
- secondary subtitle download assets are added to `VideoHD` and `VideoSD` models and are only deleted when neither HD or SD are downloaded

### CODE REVIEW

File: app/src/main/java/de/xikolo/managers/DownloadManager.kt [hunk 1]
```diff
@@ -92,21 +92,32 @@ class DownloadManager(activity: FragmentActivity) {
...
                 if (!downloadExists(downloadAsset)) {
-                    StorageUtil.cleanStorage(File(downloadAsset.filePath))
-                    false
+                    StorageUtil.cleanStorage(File(downloadAsset.filePath).parentFile)
+                    return false
                 } else {
                     EventBus.getDefault().post(DownloadDeletedEvent(downloadAsset))
-                    return File(downloadAsset.filePath).delete()
+
+                    if (getDownloadFile(downloadAsset)!!.delete()) {
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/managers/DownloadManager.kt]:
```suggestion
                    if (getDownloadFile(downloadAsset)?.delete() == true) {
```

File: app/src/main/java/de/xikolo/managers/DownloadManager.kt [hunk 2]
```diff
@@ -173,14 +188,47 @@ class DownloadManager(activity: FragmentActivity) {
...
 
-    fun getDownload(downloadAsset: DownloadAsset): Download? =
-        DownloadService.getInstance()?.getDownload(downloadAsset.url)
+    fun getDownloadTotalBytes(downloadAsset: DownloadAsset): Long {
+        var totalBytes = 0L
+        val mainDownload = DownloadService.getInstance()?.getDownload(downloadAsset.url)
+        totalBytes +=
+            if (mainDownload != null && mainDownload.totalBytes > 0L)
+                mainDownload.totalBytes
+            else downloadAsset.size
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/managers/DownloadManager.kt]: Brackets!

File: app/src/main/java/de/xikolo/managers/DownloadManager.kt [hunk 3]
```diff
@@ -173,14 +188,47 @@ class DownloadManager(activity: FragmentActivity) {
         }
     }
 
-    fun getDownload(downloadAsset: DownloadAsset): Download? =
-        DownloadService.getInstance()?.getDownload(downloadAsset.url)
+    fun getDownloadTotalBytes(downloadAsset: DownloadAsset): Long {
+        var totalBytes = 0L
+        val mainDownload = DownloadService.getInstance()?.getDownload(downloadAsset.url)
+        totalBytes +=
+            if (mainDownload != null && mainDownload.totalBytes > 0L)
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/managers/DownloadManager.kt]:
```suggestion
            if (mainDownload?.totalBytes > 0L)
```
Does this work?

File: app/src/main/java/de/xikolo/managers/DownloadManager.kt [hunk 4]
```diff
@@ -173,14 +188,47 @@ class DownloadManager(activity: FragmentActivity) {
...
+                mainDownload.totalBytes
+            else downloadAsset.size
+
+        downloadAsset.secondaryAssets.forEach {
+            val secondaryDownload = DownloadService.getInstance()?.getDownload(it.url)
+            totalBytes +=
+                if (secondaryDownload != null && secondaryDownload.totalBytes > 0L)
+                    secondaryDownload.totalBytes
+                else it.size
```
Comment (tobiasrohloff) [inline app/src/main/java/de/xikolo/managers/DownloadManager.kt]: Brackets!

File: app/src/main/java/de/xikolo/managers/DownloadManager.kt (hunk 3 above)
Comment (bensoftde) [reply app/src/main/java/de/xikolo/managers/DownloadManager.kt]:
```suggestion
            if (mainDownload != null && mainDownload.totalBytes > 0L)
```
Works only like `mainDownload?.totalBytes?.compareTo(0L)!! > 0`

**Pull request merged**